megamek/src/megamek/common/PlanetaryConditions.java

```diff
@@ -273,80 +93,123 @@ public Object clone() {
-        } else {
-            return String.valueOf(temp);
-        }
+    /** @return The time of day lighting conditions (one of PlanetaryConditions.L_*). */
+    public Light getLight() {
+        return light;
     }
 
-    public static String getWindDirDisplayableName(int type) {
-        if ((type >= 0) && (type < DIR_SIZE)) {
-            return dirNames[type];
-        }
-        throw new IllegalArgumentException("Unknown wind direction");
+    public void setWeather(Weather weather) {
+        this.weather = weather;
     }
 
-    public static String getWindDisplayableName(int type) {
-        if ((type >= 0) && (type < WI_SIZE)) {
-            return windNames[type];
-        }
-        throw new IllegalArgumentException("Unknown wind condition");
+    public Weather getWeather() {
+        return weather;
     }
 
-    public static String getAtmosphereDisplayableName(int type) {
-        if ((type >= 0) && (type < ATMO_SIZE)) {
-            return atmoNames[type];
-        }
-        throw new IllegalArgumentException("Unknown atmospheric pressure condition");
+    public void setWind(Wind wind) {
+        this.wind = wind;
     }
 
-    public static String getFogDisplayableName(int type) {
-        if ((type >= 0) && (type < FOG_SIZE)) {
-            return fogNames[type];
-        }
-        throw new IllegalArgumentException("Unknown fog condition");
+    public Wind getWind() {
+        return wind;
+    }
+
+    public void setWindMin(Wind windMin) {
+        this.windMin = windMin;
+    }
+
+    public Wind getWindMin() {
+        return windMin;
     }
 
-    public String getWindDirDisplayableName() {
-        return getWindDirDisplayableName(windDirection);
+    public void setWindMax(Wind windMax) {
+        this.windMax = windMax;
     }
 
-    public String getLightDisplayableName() {
-        return getLightDisplayableName(lightConditions);
+    public Wind getWindMax() {
+        return windMax;
     }
 
-    public String getWeatherDisplayableName() {
-        return getWeatherDisplayableName(weatherConditions);
+    public void setWindDirection(WindDirection windDirection) {
+        this.windDirection = windDirection;
     }
 
-    public String getWindDisplayableName() {
-        return getWindDisplayableName(windStrength);
+    public WindDirection getWindDirection() {
+        return windDirection;
+    }
+
+    public void setAtmosphere(Atmosphere atmosphere) {
+        this.atmosphere = atmosphere;
+    }
+
+    public Atmosphere getAtmosphere() {
+        return atmosphere;
+    }
+
+    public Fog getFog() {
+        return fog;
     }
 
-    public String getAtmosphereDisplayableName() {
-        return getAtmosphereDisplayableName(atmosphere);
+    public void setFog(Fog fog) {
+        this.fog = fog;
     }
 
-    public String getFogDisplayableName() {
-        return getFogDisplayableName(fog);
+    public BlowingSand getBlowingSand() {
```

Comment:
I suggest adding isBlowingSand() to avoid the verbose getBlowingSand().isBlowingSandNone()

Reply:
I saw that used somewhere but now I can't find it anymore